### PR TITLE
Add an integration test based on mocked objects.

### PR DESCRIPTION
**Pull request by palango**
This also fixed one bug in the pathfinding service and can be extended
later.

This is faster than full integration tests and should be used for testing edge cases.

### CODE REVIEW

File: pathfinder/tests/test_mocked_integration.py [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+        network_listener.emit_event(dict(
+            address=token_network_address,
+            name='ChannelNewDeposit',
+            args=dict(
+                channel_identifier=channel_id,
+                participant=addresses[p2_index],
+                total_deposit=p2_deposit
+            )
+        ))
+
```
Comment (err508) [inline pathfinder/tests/test_mocked_integration.py]: We should also test here, that the deposits happened right?

File: pathfinder/tests/test_mocked_integration.py [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+    #     p2_index,
+    #     p2_deposit,
+    #     p2_transferred_amount,
+    #     p2_fee
+    # )
+    # Topology:
+    #       /-------------\
+    # 0 -- 1 -- 2 -- 3 -- 4    5 -- 6
+    #  \-------/
+
```
Comment (err508) [inline pathfinder/tests/test_mocked_integration.py]: The fees are not used here. I'd either leave them out or mock the fee functionality to make the test more integrational.

File: pathfinder/tests/test_mocked_integration.py [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+            name='ChannelClosed',
+            args=dict(
+                channel_identifier=channel_id,
+                closing_participant=addresses[p1_index]
+            )
+        ))
+
+    # there should be no channels
+    assert len(token_network.channel_id_to_addresses.keys()) == 0
+
```
Comment (err508) [inline pathfinder/tests/test_mocked_integration.py]: For an integration test, I'd suggest to also test if the `get_paths` method works, as it is the core functionality of the PFS.

File: pathfinder/tests/test_mocked_integration.py [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+            args=dict(
+                channel_identifier=1,
+                participant1=addresses[0],
+                participant2=addresses[1]
+            )
+        ))
+
+    # now there should be seven channels
+    assert len(token_network.channel_id_to_addresses.keys()) == 1
+
```
Comment (err508) [inline pathfinder/tests/test_mocked_integration.py]: now there should be only _one_ channel

**Comment by err508** [comment]
I'd maybe change the commit message and the test descriptions a little to introduce the notion of integration_depth, i.e. to make clear at what level stuff gets mocked. Something like **integration test, integration_depth = blockchain_listener**  or so. Than one can easily see the difference between integration_depth = blockchain_listener and integration_depth = raiden_client. I'd name the test I'm working on respectively.

File: pathfinder/tests/test_mocked_integration.py (hunk 2 above)
Comment (palango) [reply pathfinder/tests/test_mocked_integration.py]: I'd keep them for the future and integrate this as soon as the message format is known.

File: pathfinder/tests/test_mocked_integration.py (hunk 1 above)
Comment (palango) [reply pathfinder/tests/test_mocked_integration.py]: Done

File: pathfinder/tests/test_mocked_integration.py (hunk 3 above)
Comment (palango) [reply pathfinder/tests/test_mocked_integration.py]: Done

**Comment by palango** [comment]
Added some docs that describe to what level this is mocked, also renames the test cases.
